pkg/helm/v2/repository.go

```diff
@@ -0,0 +1,142 @@
+package v2
+
+import (
+	"os"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/helm/pkg/getter"
+	"k8s.io/helm/pkg/repo"
+)
+
+var (
+	repositoryConfigLock sync.RWMutex
```

Comment:
is there a good reason to have this as a global variable? Can't it be part of `Helm2`?

Reply:
There can theoretically be multiple versions of `Helm2` while they all speak with the same lock file (this is not completely true, as they can have different `HELM_HOME`s and are thus working with a different file, but this is the simplest secure option).